coverage map branch cleanup

arneboe opened the pull request:
This PR contains all commits that where on the coverage_map branch but had nothing to do with the coverage_map. Mostly bugfixes and some small features.

chhtz committed:
Don't do BOOST_CHECK_EQUAL to boolean variables

chhtz committed:
introduce toGridLocal and fromGridLocal, which ignore the LocalFrame

chhtz committed:
fix warning

chhtz committed:
fix warning

chhtz committed:
use boost::flat_map instead of std::map for better performance

chhtz committed:
Variant of fromGridLocal, which just takes a scalar height

chhtz committed:
maxIdx must be rounded up, to make sure that partially intersecting patches are included

Commit:
getClosestNode() added

chhtz committed:
Sanity check

arneboe committed:
add getMin to TravMap

arneboe committed:
Add overlap that works with pointers

Commit:
fixed memory management of TraversabilityMap3d

The map now gains control of the node given to it.
On deletion all nodes held by the map are deleted.
The copy operator / constructor now generate deep
copies.
For performance reasons the move constructor and
operator have been implemented.

saarnold commented:
There are two changes to existing map types involved in those comments.
Currently this would break backward compatibility with serialization for those types.
We should check if we can add support for those changes, before we merge this.

saarnold commented:
@JanWehrmann will implement backward compatibility.

JanWehrmann committed:
Added backwards compatibility for DiscreteTree
and tests to ensure backwards compatibility of
DiscreteTree and TraversabilityMap3d.

saarnold committed:
Merge pull request #25 from JanWehrmann/branch_cleanup

Added backwards compatibility for DiscreteTree
